synapse/lib/stormlib/iters.py

```diff
@@ -27,10 +41,34 @@ def __init__(self, runt, name=()):
     def getObjLocals(self):
         return {
             'enum': self.enum,
+            'zip': self._zip,
         }
 
     async def enum(self, genr):
         indx = 0
         async for item in s_stormtypes.toiter(genr):
             yield (indx, item)
             indx += 1
+
+    async def _zip(self, *args):
+
+        async with contextlib.AsyncExitStack() as stack:
+            genrs = []
+            for arg in args:
+                agen = contextlib.aclosing(s_stormtypes.toiter(arg))
+                genrs.append(await stack.enter_async_context(agen))
+
+            try:
+                while True:
+                    tasks = []
+                    async with asyncio.TaskGroup() as tg:
```

Comment:
This doesn't appear to handle generators with different lengths like the python implementation does. Am i missing it? We may have to use some spooled iters magic here :)

Reply:
it handles them the same way as python does, stopping when the shortest is exhausted